lib/extras/codec_png.cc

```diff
@@ -569,8 +571,16 @@ class ColorEncodingWriterPNG {
 
   static Status MaybeAddGAMA(const ColorEncoding& c,
                              LodePNGInfo* JXL_RESTRICT info) {
-    if (!c.tf.IsGamma()) return true;
-    const double gamma = c.tf.GetGamma();
+    double gamma;
+    if (c.tf.IsGamma()) {
+      gamma = c.tf.GetGamma();
+    } else if (c.tf.IsLinear()) {
+      gamma = 1;
+    } else if (c.tf.IsSRGB()) {
+      gamma = 0.45455;
```

Comment:
This is only an approximation to sRGB, would you say it's useful to add this at all?

Fortunately, sRGB and ICC chunk override the gama chunk, so it doesn't hurt at least, so I'm ok to keep

Reply:
Yep :) and more generally:

> An application that writes the iCCP chunk should also write gAMA and cHRM chunks that approximate the ICC profile's transfer function, for compatibility with applications that do not use the iCCP chunk.

So I think this value should be fine even for colorspaces that are not sRGB but only share its TRC (e.g. Display P3).